src/services/side-panels/side-panels-manager.tsx

```diff
@@ -0,0 +1,184 @@
+import {h, createRef, RefObject, FunctionalComponent, ComponentClass} from 'preact';
+import {ui, KalturaPlayer, Logger} from 'kaltura-player-js';
+import {ISidePanelItemDto, SidePanelItemDto} from './side-panel-item-dto';
+import {Toggle} from './ui/side-panel.component';
+import {SidePanelPosition} from './types/types';
+import {ItemWrapper} from './item-wrapper';
+
+const {SidePanelModes, SidePanelPositions, ReservedPresetNames, ReservedPresetAreas} = ui;
+
+const OPPOSITE_PANELS: Record<SidePanelPosition, SidePanelPosition> = {
+  [SidePanelPositions.TOP]: SidePanelPositions.BOTTOM,
+  [SidePanelPositions.BOTTOM]: SidePanelPositions.TOP,
+  [SidePanelPositions.RIGHT]: SidePanelPositions.LEFT,
+  [SidePanelPositions.LEFT]: SidePanelPositions.RIGHT,
+} as Record<SidePanelPosition, SidePanelPosition>;
+
+/**
+ * Class representing a socket connection.
+ *
+ * @class
+ */
+export class SidePanelsManager {
+  private readonly player: KalturaPlayer;
+  private readonly activePanels: Record<SidePanelPosition, ItemWrapper | null>;
+  private readonly componentsRegistry: Map<number, ItemWrapper>;
+  private readonly logger: Logger;
+
+  constructor(player: KalturaPlayer, logger: Logger) {
+    this.player = player;
+    this.activePanels = {top: null, bottom: null, right: null, left: null};
+    this.componentsRegistry = new Map<number, ItemWrapper>();
+    this.logger = logger;
+  }
+
+  public addItem(item: ISidePanelItemDto): number {
+    if (SidePanelsManager.validateItem(item)) {
+      const newPanelItem: SidePanelItemDto = new SidePanelItemDto(item);
+      const {componentRef, removeComponentFunc} = this.injectPanelComponent(item);
+      const newItemWrapper: ItemWrapper = new ItemWrapper(newPanelItem, componentRef, removeComponentFunc);
+      if (item.renderIcon) this.injectIconComponent(newItemWrapper);
+      this.componentsRegistry.set(newItemWrapper.id, newItemWrapper);
+      this.logger.debug('1234 New Panel Item Added', item);
+      return newItemWrapper.id;
+    }
+    this.logger.warn('invalid SidePanelItem parameters', item);
+    throw new Error('invalid SidePanelItem parameters');
+  }
+
+  public removeItem(itemId: number): void {
+    const item: ItemWrapper | undefined = this.componentsRegistry.get(itemId);
+    if (item) {
+      this.deactivateItem(itemId);
+      item.removeComponentFunc();
```

Comment:
whats about the icon? where is its removal?

Reply:
fixed